dirsync/run.py

```diff
@@ -8,6 +8,7 @@
 import os
 
 from .syncer import Syncer
+from .options import ArgParser, USER_CFG_FILE, DEFAULT_USER_CFG
 

```

Comment:
The reason the options module was only imported within `from_cmdline` is that it has side effects (creating the options dictionary + arg parser class, etc.), which is not needed at all when calling dirsync from python. That's why I would prefer only importing it from within `from_cmdline`. If the tests are still needed, they could import `ArgParser` from the `options` module.

Reply:
personal I do not like import in functions. I like to have all the import at the top of the file where I would expect them to find. When I look at the options module, I think having the import of the options module only for cmd_line scope, would be a premature optimization.

Reply:
I do not like imports in functions either and endeavour to put all imports at the top, but in a few cases like this one they make sense. Imports in python are pretty expensive (especially as the options module imports other modules) so if it is easy to avoid importing modules we know we won't need in some well-defined use cases, we do it.

A change that is trivial to implement and understand and can save 10s of ms each time a module is imported is not 'premature optimisation' (a term that has been debased), it's just common sense!

If we really wanted to avoid imports in functions, you could have create a separate `from_cmdline` module that would import the bits it needs from `options`, but as `from_cmdline` and `run` essentially do the same thing, it makes a lot of sense to keep them in the same module.

Would you mind reverting that change please? Thanks in advance.